roofit/hs3/src/JSONFactories_RooFitCore.cxx

```diff
@@ -98,37 +169,139 @@ class RooSimultaneousFactory : public RooJSONFactoryWSTool::Importer {
+
+      if (!p.has_child("epsilon")) {
+         RooJSONFactoryWSTool::error("no epsilon given in '" + name + "'");
+      }
+      double epsilon(p["epsilon"].val_float());
+
+      RooBinSamplingPdf thepdf(name.c_str(), name.c_str(), *obs, *pdf, epsilon);
+      tool->workspace()->import(thepdf, RooFit::RecycleConflictNodes(true), RooFit::Silence(true));
+
       return true;
    }
 };
+} // namespace
+
+#include <RooRealSumPdf.h>
+
+namespace {
+class RooRealSumPdfFactory : public RooJSONFactoryWSTool::Importer {
+public:
+   virtual bool importPdf(RooJSONFactoryWSTool *tool, const JSONNode &p) const override
+   {
+      std::string name(RooJSONFactoryWSTool::name(p));
+      if (!p.has_child("samples")) {
+         RooJSONFactoryWSTool::error("no samples given in '" + name + "'");
+      }
+      if (!p.has_child("coefficients")) {
+         RooJSONFactoryWSTool::error("no coefficients given in '" + name + "'");
+      }
+      RooArgList samples;
+      for (const auto &sample : p["samples"].children()) {
+         RooAbsReal *s = tool->request<RooAbsReal>(sample.val(), name);
+         samples.add(*s);
+      }
+      RooArgList coefficients;
+      for (const auto &coef : p["coefficients"].children()) {
+         RooAbsReal *c = tool->request<RooAbsReal>(coef.val(), name);
+         coefficients.add(*c);
+      }
 
+      bool extended = false;
+      if (p.has_child("extended") && p["extended"].val_bool()) {
+         extended = true;
+      }
+      RooRealSumPdf thepdf(name.c_str(), name.c_str(), samples, coefficients, extended);
+      tool->workspace()->import(thepdf, RooFit::RecycleConflictNodes(true), RooFit::Silence(true));
+      return true;
+   }
+};
 } // namespace
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
 // specialized exporter implementations
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
 
+#include <RooRealSumPdf.h>
+
+namespace {
+class RooRealSumPdfStreamer : public RooJSONFactoryWSTool::Exporter {
+public:
+   virtual std::string key() const { return "sumpdf"; }
+   virtual bool exportObject(RooJSONFactoryWSTool *, const RooAbsArg *func, JSONNode &elem) const override
+   {
+      const RooRealSumPdf *pdf = static_cast<const RooRealSumPdf *>(func);
+      elem["type"] << key();
+      auto &samples = elem["samples"];
+      samples.set_seq();
+      auto &coefs = elem["coefficients"];
+      coefs.set_seq();
+      for (const auto &s : pdf->funcList()) {
+         samples.append_child() << s->GetName();
+      }
+      for (const auto &c : pdf->coefList()) {
+         coefs.append_child() << c->GetName();
+      }
+      elem["extended"] << (pdf->extendMode() == RooAbsPdf::CanBeExtended);
+      return true;
+   }
+};
+} // namespace
+
 namespace {
 class RooSimultaneousStreamer : public RooJSONFactoryWSTool::Exporter {
 public:
+   virtual std::string key() const { return "simultaneous"; }
```

Comment:
```suggestion
   virtual std::string const& key() const {
      const static std::string keystring = "simultaneous";   
      return keystring;      
}
```
If you always return the same string anyway, is there not quite a bit of overhead when you always create a new string? What about changing this `key` interface to return references to static strings?